src/data/sym/basic.lean

```diff
@@ -237,13 +270,51 @@ by simp [sym.map]
   (a :: s).map f = (f a) :: s.map f :=
 by simp [map, cons]
 
+@[congr] lemma map_congr {β : Type*} {f g : α → β} {s : sym α n} (h: ∀ x ∈ s, f x = g x) :
+  map f s = map g s := subtype.ext $ multiset.map_congr rfl h
+
+@[simp] lemma map_mk {β : Type*} {f : α → β} {m : multiset α} {hc : m.card = n} :
+  map f (mk m hc) = mk (m.map f) (by simp [hc]) := rfl
+
+@[simp] lemma coe_map {β : Type*} (s : sym α n) (f : α → β) : ↑(s.map f) = multiset.map f s := rfl
+
+lemma map_injective {β : Type*} {f : α → β} (hf : injective f) (n : ℕ) :
+  injective (map f : sym α n → sym β n) :=
+λ s t h, coe_injective $ multiset.map_injective hf $ coe_inj.2 h
+
 /-- Mapping an equivalence `α ≃ β` using `sym.map` gives an equivalence between `sym α n` and
 `sym β n`. -/
 @[simps]
-def equiv_congr {β : Type u} (e : α ≃ β) : sym α n ≃ sym β n :=
+def equiv_congr {α β : Type*} (e : α ≃ β) : sym α n ≃ sym β n :=
 { to_fun := map e,
   inv_fun := map e.symm,
   left_inv := λ x, by rw [map_map, equiv.symm_comp_self, map_id],
   right_inv := λ x, by rw [map_map, equiv.self_comp_symm, map_id] }
 
+/-- "Attach" a proof that `a ∈ s` to each element `a` in `s` to produce
+an element of the symmetric power on `{x // x ∈ s}`. -/
+def attach (s : sym α n) : sym {x // x ∈ s} n := ⟨s.val.attach, by rw [multiset.card_attach, s.2]⟩
+
+@[simp] lemma attach_mk {m : multiset α} {hc : m.card = n} :
+  attach (mk m hc) = mk m.attach (by simp [hc]) := rfl
+
+@[simp] lemma coe_attach (s : sym α n) : (s.attach : multiset {a // a ∈ s}) = multiset.attach s :=
+rfl
+
+lemma attach_map_coe (s : sym α n) : s.attach.map coe = s :=
+coe_injective $ multiset.attach_map_val _
+
+@[simp] lemma mem_attach : Π (s : sym α n), ∀ x, x ∈ s.attach
+| (mk s _) := by simp
```

Comment:
```suggestion
@[simp] lemma mem_attach (s : sym α n) (x : {x // x ∈ s}) : x ∈ s.attach :=
multiset.mem_attach _ _
```
Proving things in term of the `multiset` versions is a nice way to confirm the argument explicitness matches.